Source/HTTP/Curl/CurlEasyRequest.cpp

```diff
@@ -38,17 +39,14 @@ Result<HC_UNIQUE_PTR<CurlEasyRequest>> CurlEasyRequest::Initialize(HCCallHandle
     void* clientRequestBodyReadCallbackContext{};
     RETURN_IF_FAILED(HCHttpCallRequestGetRequestBodyReadFunction(hcCall, &clientRequestBodyReadCallback, &bodySize, &clientRequestBodyReadCallbackContext));
 
-    if (bodySize > 0)
-    {
-        // we set both POSTFIELDSIZE and INFILESIZE because curl uses one or the
-        // other depending on method
-        RETURN_IF_FAILED(easyRequest->SetOpt<long>(CURLOPT_POSTFIELDSIZE, static_cast<long>(bodySize)));
-        RETURN_IF_FAILED(easyRequest->SetOpt<long>(CURLOPT_INFILESIZE, static_cast<long>(bodySize)));
-
-        // read callback
-        RETURN_IF_FAILED(easyRequest->SetOpt<curl_read_callback>(CURLOPT_READFUNCTION, &ReadCallback));
-        RETURN_IF_FAILED(easyRequest->SetOpt<void*>(CURLOPT_READDATA, easyRequest.get()));
-    }
+    // we set both POSTFIELDSIZE and INFILESIZE because curl uses one or the
```

Comment:
why?

Reply:
It was to resolve a bug I found with the curl library. This doesn't happen in xcurl. What happened was that when creating a WebSocket it would never complete and that was because MLP wanted a websocket it a fixed body. When sending the request for the handshake it would send no data and these setOpts would never be initialized. Removing the if statement solves the issue. Bodysize is unsigned so I we don't have to worry about a negative number.

Reply:
@SebastianPD do we know that this is actually a bug in curl? or could this just be curl's (perhaps poorly documented) intentional behavior? 

in general, the goal of xcurl is to mirror curl's behavior as closely as reasonably possible. If this isn't definitely a bug it's possible there's reasons xcurl should mirror this as well

> beacuse MLP wanted a websocket it a fixed body
can you elaborate on this? if the bodySize here is zero, what additional data is it now sending?

additionally, have we validated that the requests sent by xcurl haven't changed at all by removing this if-check? even if things still work it'd be interesting to know whether the requests are sent with any additional headers which might not have been there beforeor if the CURLOPT_READFUNCTION is now called in a code path in which it previously wasn't

Reply:
> because MLP wanted a webscoket it a fixed body

I also wonder whether this is a bug in MLP instead of libhttpclient. What requires the fixed body and could be address it at the MLP level?

(if discussing this in any meaningful way requires sharing internal details about MLP let's chat out-of-band, maybe on teams. we should avoid putting MLP internal code in this public repo even in comments)

Reply:
what would happen previously when doing the handshake was `Transfer-Encoding: chunked` was a header and removing this if statement and setting the set opt would make the header say it has a size of 0. The body would have nothing in it. Those were the only differences when looking at the fiddler trace.